relation/index.js

```diff
@@ -108,7 +111,39 @@ module.exports = yeoman.generators.Base.extend({
           }
           return m;
         },
-        validate: validateName
+        validate: function(value) {
+
+            var done = this.async();
+
+            //validfn returns false if relationship 
+            //has the same name with properties
+            var validfn = function(value, cb) {
```

Comment:
Well, this may be not necessary after you apply refactorings that I suggested below.